source/elements/oneDAL/source/data_management/index.rst

```diff
@@ -5,12 +5,294 @@
+:ref:`data analytics pipeline <data_analytics_pipeline>` or how to access its
+data).
+
+Metadata for all datasets shall contain:
+
+- The number of rows :math:`n` and columns :math:`p` in a dataset
+
+- Type of each :term:`feature` (e.g. :term:`nominal <Nominal feature>`,
+  :term:`interval <Interval feature>`)
+
+- :term:`data type` of each feature
+
+- The kind of a dataset (e.g. :ref:`table` or :ref:`data-source`)
+
+.. note::
+  Metadata can contain both compile-time and run-time information. For example,
+  basic compile-time metadata is the type of a dataset - whether it is a
+  particular :ref:`data-source` or a :ref:`table`. Run-time information can
+  contain the :term:`feature` types and :term:`data types <Data type>` of a
+  dataset.
+
+.. note::
+  Each concept that concretizes a dataset definition (like different tables or
+  data sources) shall contain additional metadata which describes this concept
+  as the whole. For example, :ref:`data-source` and :ref:`table` concepts have
+  different sets of metadata since they represent different kinds of a
+  :term:`dataset`.
+
+.. _table-builder:
+
+Table builder
+-------------
+
+A table :term:`builder` is a concept that associated with particular
+:ref:`table` type and is used on data-preparation and data-processing stages
+for:
+
+- Iterative (in general case) construction of a :ref:`table` from
+  another :ref:`tables <table>` or different in-memory :ref:`dataset`
+  representations
+
+- Construction of a :ref:`table` from different entities that hold pieces of the
+  dataset, like arrays, pointers to the memory, external entities, etc.
+
+- Changing dataset values. Since :ref:`table` is an
+  :term:`immutable <Immutability>` dataset, builder provides an ability to
+  change the values of data in the dataset under construction.
+
+- Incapsulating construction process of a :ref:`table`. This is used to hide an
+  implementation details, which (1) are unnecessary for user, (2) allow to
+  select most appropriate table implementation for every particular case.
+
+- Providing an additional information how to create a :ref:`table` inside an
+  algorithm for :ref:`results  <Result>`. This information may include metadata,
+  memory allocators need to be used, or even particular implementation of a
+  table.
+
+For details, see :ref:`table-builders` section.
+
+.. _accessor:
+
+Accessor
+--------
+
+Accessor is a concept which defines sinle way how to get the data from a
+in-memory numerical :ref:`dataset`. It allows:
+
+- To make uniform access to the data from various sets of different objects,
+  like :ref:`tables <Table>` or :ref:`table builders <table-builder>` without
+  exposing their implementation details.
+
+- To convert a variety of numeric :term:`data formats <Data format>` into a
+  smaller set (e.g. for better vectorization).
+
+- To give an ability to focus on the data acquisition in the desired :term:`data
+  format` with a wanted set of operations over the data.
```

Comment:
already commented on this: https://github.com/oneapi-src/oneAPI-spec/pull/141#discussion_r430404995